issue142 : Add Divider to EditProfile, theme styles, and custom Divider component

Student-ParasKapoor opened the pull request:
### Why is this change needed?

This PR introduces the `Divider` component to ensure consistent usage of dividers across the app with support for orientation, dash style, thickness, length, and theme-aware colors. It enforces design system abstraction by restricting color input to predefined tokens.

### What has changed?

- Implemented `Divider.tsx` component
- Created `Divider.styles.ts` for dynamic orientation and dash styles
- Restricted `color` prop to design tokens like `primary`, `neutral`, etc.
- Used `useThemeColor` hook to safely resolve colors from the theme
- Added unit tests in `divider.test.tsx`
- Updated usage in `edit-profile.tsx` to use the new `Divider`

### API Changes

- `DividerProps` includes orientation, thickness, color (as token), length, dashStyle, and style
- `color` accepts only design system tokens (e.g. `primary`, `neutral`)

### Database Changes

- None

### Screenshots

N/A (non-visual or minor visual UI component)

### Manual Testing

- Verified in `edit-profile` screen with all supported props
- Tested all orientations, lengths, colors, dash styles
- Validated safe fallback behavior for undefined or invalid colors

### Automated Testing

- ✅ `divider.test.tsx` includes tests for all supported props and visual styles
- ✅ All tests pass locally

### Additional Notes

- The `useThemeColor` hook is now type-safe and protects against invalid theme access
- Component follows accessibility best practices (excluding `accessibilityRole="separator"` due to Android issues)

jjalan commented on .eslintrc.js:
```diff
@@ -27,6 +27,14 @@ module.exports = {
   settings: {
     'import/resolver': {
       'babel-module': {},
+      node: {
```
Can you explain why these changes are needed, and more so why are they needed in this PR?

jjalan commented on package.json:
```diff
@@ -71,6 +71,7 @@
     "dotenv": "^16.3.1",
     "eslint": "^8.19.0",
     "eslint-import-resolver-babel-module": "^5.3.2",
+    "eslint-import-resolver-typescript": "^4.4.3",
```
Why is this needed?

jjalan commented on src/app-theme.ts:
```diff
@@ -171,7 +171,19 @@ const appTheme = extendTheme({
 });
 
-export default appTheme;
+export default appTheme;
```
Here and other files, end of line is missing

jjalan commented on src/app-theme.ts:
```diff
@@ -171,7 +171,19 @@ const appTheme = extendTheme({
     },
+    Divider: {
+      baseStyle: {
+        bg: 'secondary.200', // Light mode color (#E5E5E5)
```
Avoid one liner comment, see guidance on how to write comments: [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)

jjalan commented on src/components/divider/Readme.md:
File name should be case insensitive, rename it to `readme.md`

jjalan commented on src/components/divider/divider.styles.ts:
```diff
@@ -0,0 +1,28 @@
+  thickness: number;
+  length?: number | string;
+  dashStyle?: 'solid' | 'dashed' | 'dotted';
+  dividerColor: string;
```
We do not want to let consumer pass any color they choose. This interface allows consumer to pass any color and break the abstraction. Potentially look at open source libs or work with design to build abstractions (maybe with kind such as primary, secondary etc)

jjalan commented on src/components/divider/divider.styles.ts:
```diff
@@ -0,0 +1,28 @@
+  orientation: 'horizontal' | 'vertical';
+  thickness: number;
+  length?: number | string;
+  dashStyle?: 'solid' | 'dashed' | 'dotted';
```
add enum for dash style for type safety

jjalan commented on src/components/divider/divider.styles.ts:
```diff
@@ -0,0 +1,28 @@
+interface DividerStyleProps {
+  orientation: 'horizontal' | 'vertical';
+  thickness: number;
+  length?: number | string;
```
Why string?

jjalan commented on src/components/divider/divider.styles.ts:
```diff
@@ -0,0 +1,28 @@
+import { StyleSheet, ViewStyle } from 'react-native';
+
+interface DividerStyleProps {
+  orientation: 'horizontal' | 'vertical';
```
create an enum for typesafety

jjalan commented on src/components/divider/divider.test.tsx:
```diff
@@ -0,0 +1,167 @@
+
+import Divider, { DividerProps } from './divider';
+
+// Mock the useThemeColor hook
```
Avoid comments like this, here and other places in this PR, see: [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)

jjalan commented on src/components/divider/divider.tsx:
```diff
@@ -0,0 +1,53 @@
+
+import { useDividerStyles } from './divider.styles';
+
+/**
```
Remove comments, code should be readable on its own

jjalan commented on yarn.lock:
There should not be any change in this file.

Student-ParasKapoor committed:
Normalize line endings to LF

Student-ParasKapoor committed:
Add Divider component and update use-theme-color
- Added src/components/divider/ (divider.tsx, divider.styles.ts, divider.test.tsx, index.ts)
- Updated src/utils/use-theme-color.hook.ts for Divider theming
- Added src/utils/.gitattributes (if kept)

jjalan commented on src/app-theme.ts:
```diff
@@ -26,28 +128,28 @@ const ICOLORHUES = {
-    700: '#be185d',
-    800: '#9d174d',
-    900: '#831843',
+    '50': '#fdf2f8',
```
Why are we changing this?

jjalan commented on android/gradlew.bat:
```diff
@@ -1,92 +1,92 @@
-if "%OS%"=="Windows_NT" endlocal
-
-:omega
+@rem
```
What changed in this file and why is the change needed?

jjalan commented on src/utils/use-theme-color.hook.ts:
```diff
@@ -1,44 +1,26 @@
   return useMemo(() => {
-    const [colorKey, shade = '500'] = color.split('.');
-    const themeColor = theme.colors[colorKey as keyof ITheme['colors']];
+    const [colorKey, shade = '200'] = color.split('.') as [ThemeColorKey, string];
```
why change from 500 to 200 default?

jjalan reviewed (commented):
There are quite a few unwanted changes in this PR which is not desirable.

Kindly self review the PR and make sure only divider related component changes exists in the PR
